py-polars/tests/unit/operations/test_replace.py

```diff
@@ -493,6 +493,36 @@ def test_replace_fast_path_many_to_one_null() -> None:
     assert_frame_equal(result, expected)
 
 
+def test_replace_duplicates_old_single() -> None:
+    s = pl.Series([1, 2, 3, 2, 3])
+    result = s.replace([2, 2], 100)
```

Comment:
I think we should raise here? Why would we allow duplicates here? It is in `old` right?

Reply:
It's not ambigous so we _can_ allow it. This is the fast path and it'll just do `s.is_in([2, 2])` which is a valid query.

It gets more dicey when doing `replace([1, 1], [2, 2])` which requires deduplication before proceeding - which I added in this PR. Though we could also choose to raise to save some computation.

`replace([1, 1], [2, 3])` is ambiguous and _must_ raise.

Reply:
I understand that it works. But i think we should be strict here. This can prevent future optimizations and it just feels like bad input to me.

Reply:
But isn't that the responsibility of `is_in` then, to raise on bad inputs?

I can simply do a polars ensure with `is_unique(old).all()` though if you think that's better.

Reply:
I think we should check that. Fastest is probably `n_unique() == len()`